controller/eks-cluster-config-handler.go

```diff
@@ -1046,28 +1045,33 @@ func (h *Handler) updateUpstreamClusterState(upstreamSpec *eksv1.EKSClusterConfi
 		if _, ok := upstreamNgs[aws.StringValue(ng.NodegroupName)]; ok {
 			continue
 		}
-		_, err := h.awsServices.ec2.DescribeLaunchTemplates(&ec2.DescribeLaunchTemplatesInput{
-			LaunchTemplateIds: []*string{aws.String(config.Status.ManagedLaunchTemplateID)},
-		})
-		if config.Status.ManagedLaunchTemplateID == "" || doesNotExist(err) {
-			lt, err := createLaunchTemplate(config.Spec.DisplayName, ec2Service)
-			if err != nil {
-				return config, err
-			}
-			config.Status.ManagedLaunchTemplateID = aws.StringValue(lt.ID)
-		} else if err != nil {
-			return config, err
+		if err := awsservices.CreateLaunchTemplate(awsservices.CreateLaunchTemplateOptions{
+			EC2Service: h.awsServices.ec2,
+			Config:     config,
+		}); err != nil {
+			return config, fmt.Errorf("error getting or creating launch template: %w", err)
 		}
 		// in this case update is set right away because creating the
 		// nodegroup may not be immediate
 		if config.Status.Phase != eksConfigUpdatingPhase {
 			config.Status.Phase = eksConfigUpdatingPhase
-			config, err = h.eksCC.UpdateStatus(config)
+			config, err := h.eksCC.UpdateStatus(config)
 			if err != nil {
 				return config, err
 			}
 		}
-		ltVersion, generatedNodeRole, err := createNodeGroup(config, ng, eksService, ec2Service, svc)
```

Comment:
I think that, after deleting this line and changing `createNodeGroup`, the `svc` variable is no longer used in `updateUpstreamClusterState` and can be removed from the input parameters in https://github.com/rancher/eks-operator/blob/d4b228004290eec52af0efc8da8e0eaa37ea0060/controller/eks-cluster-config-handler.go#LL919C20-L919C20